code/datums/components/burning.dm

```diff
@@ -63,6 +72,35 @@ GLOBAL_DATUM_INIT(fire_overlay, /mutable_appearance, mutable_appearance('icons/e
 
 	examine_list += span_danger("[source.p_theyre(TRUE)] burning!")
 
+/// Handles searing the hand of anyone who tries to touch parent without protection.
+/datum/component/burning/proc/on_attack_hand(atom/source, mob/living/carbon/user)
+	SIGNAL_HANDLER
+
+	var/can_handle_hot = FALSE
+	if(!iscarbon(user))
+		can_handle_hot = TRUE
+	else if(user.gloves && (user.gloves.max_heat_protection_temperature >= BURNING_ITEM_MINIMUM_TEMPERATURE))
+		can_handle_hot = TRUE
+	else if(HAS_TRAIT(user, TRAIT_RESISTHEAT) || HAS_TRAIT(user, TRAIT_RESISTHEATHANDS))
+		can_handle_hot = TRUE
+	else if((source == user) || (source.loc == user)) // So people can take their own clothes off.
+		can_handle_hot = TRUE
+
+	if(can_handle_hot)
```

Comment:
smells like something that should be a proc which returns true/false

Reply:
yep sure ok will do